### PR TITLE
local-gadget: Run snapshot gadgets using tracers instead of the Tracer CR

### PR DESCRIPTION
**Pull request by blanquicet**
# Run snapshot gadgets using tracers instead of the Tracer CR

This PR fixes https://github.com/kinvolk/inspektor-gadget/issues/862.

## Implementation details
- I reverted all the modifications introduced in [#730](https://github.com/kinvolk/inspektor-gadget/pull/730) to run the snapshot gadgets using the Tracer CR.
- I moved all the TraceCR-related code back from the `common` folder to the `kubectl-gadget` folder as it is not going to be used anymore by `local-gadget`.
- I implemented the logic for iterating over network namespaces in the socket gadget, but I keep it disabled until [#744](https://github.com/kinvolk/inspektor-gadget/issues/744) is implemented. Otherwise, we will show a list of sockets without any information about the container where they are running.

## How to use

See new [process](https://github.com/kinvolk/inspektor-gadget/blob/jose/use-tracers-for-snapshot/docs/local-gadget.md#snapshotprocess) examples.

### CODE REVIEW

File: cmd/local-gadget/snapshot/socket.go [hunk 1]
```diff
@@ -15,54 +15,79 @@
...
-		"pod",
-		"protocol",
-		"local",
-		"remote",
-		"status",
-		"inode",
-	}
+				// Given that the tracer works per network namespace, we only
+				// need to run it once per namespace.
+				visitedNetNs := make(map[string]struct{})
```
Comment (mauriciovasquezbernal) [inline cmd/local-gadget/snapshot/socket.go:45]: Can you use uint14 as the map key?

**Review by mauriciovasquezbernal — APPROVED** [review]
LGTM. Just a minor comment but otherwise it can go in IMO.

File: cmd/local-gadget/snapshot/socket.go (hunk 1 above)
Comment (blanquicet) [reply cmd/local-gadget/snapshot/socket.go:45]: You are right. There is no reason to transform it into a string.

Beyond this, I think it is a pity to have this gadget disabled, everything is in place to add `local-gadget snapshot socket`, but we need to solve https://github.com/kinvolk/inspektor-gadget/issues/744. Maybe @alban is interested in working on it?

**Comment by blanquicet** [comment]
Rebased to solve conflicts.

**Pull request merged**